graphql-dgs-graphiql-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLPathTestApplication.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.graphiql.autoconfiguration
+
+import com.netflix.graphql.dgs.DgsQueryExecutor
+import com.netflix.graphql.dgs.internal.DgsSchemaProvider
+import io.mockk.mockk
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.context.annotation.Bean
+
+@SpringBootApplication
+open class GraphiQLPathTestApplication {
```

Comment:
Just curious where this is being used.

Reply:
Ahh yes, I should have commented about this. Good catch. The **TL;DR** is <img src="https://user-images.githubusercontent.com/28907971/111069114-93d4ed80-8499-11eb-9be5-51c8e9945149.png" width=50/> magic.

If you comment out the `@SpringBootApplication` on `GraphiQLPathTestApplication` you will see failures like this: 

```java
Unable to find a @SpringBootConfiguration, you need to use @ContextConfiguration or @SpringBootTest(classes=...) with your test
java.lang.IllegalStateException: Unable to find a @SpringBootConfiguration, you need to use @ContextConfiguration or @SpringBootTest(classes=...) with your test
	at org.springframework.util.Assert.state(Assert.java:76)
	at org.springframework.boot.test.context.SpringBootTestContextBootstrapper.getOrFindConfigurationClasses(SpringBootTestContextBootstrapper.java:235)
	at org.springframework.boot.test.context.SpringBootTestContextBootstrapper.processMergedContextConfiguration(SpringBootTestContextBootstrapper.java:151)
	at org.springframework.test.context.support.AbstractTestContextBootstrapper.buildMergedContextConfiguration(AbstractTestContextBootstrapper.java:395)
```

The reason is:

> When testing Spring Boot applications, this is often not required. Spring Boot’s @*Test annotations search for your primary configuration automatically whenever you do not explicitly define one.
> 
> The search algorithm works up from the package that contains the test until it finds a class annotated with @SpringBootApplication or @SpringBootConfiguration. As long as you structured your code in a sensible way, your main configuration is usually found.
> 

Taken from [here](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-testing-spring-boot-applications-detecting-config). 

While we could specify it explicitly w/ `@SpringBootApplication(classes = [ ....`, that mechanism disables the discovery of other auto-configs and results in having to list more auto-configs in that `classes` attribute. 

I added a comment.